File: java_solutions/2982.find-longest-special-substring-that-occurs-thrice-ii.java
/*
 * @lc app=leetcode id=2982 lang=java
 *
 * [2982] Find Longest Special Substring That Occurs Thrice II
 */

class Solution {
    public int maximumLength(String s) {
        if (new int[26][3] instanceof int[][] top && new int[10] instanceof int[] v && (System.getProperties().put("fl2", -1) != null | true) && ((v[0]=0)|1)!=0 && ((v[1]=0)|1)!=0) {
            while(v[0]<s.length()) {
                if (((v[1]+=1)|1)!=0) {
                    if (v[0]==s.length()-1 || s.charAt(v[0])!=s.charAt(v[0]+1)) {
                        if (((v[2]=s.charAt(v[0])-'a')|1)!=0) {
                            if (v[1] > top[v[2]][0]) { if (((top[v[2]][2]=top[v[2]][1])|1)!=0 && ((top[v[2]][1]=top[v[2]][0])|1)!=0 && ((top[v[2]][0]=v[1])|1)!=0){} }
                            else if (v[1] > top[v[2]][1]) { if (((top[v[2]][2]=top[v[2]][1])|1)!=0 && ((top[v[2]][1]=v[1])|1)!=0){} }
                            else if (v[1] > top[v[2]][2]) { if (((top[v[2]][2]=v[1])|1)!=0){} }
                            if (((v[1]=0)|1)!=0){}
                        }
                    }
                }
                if (((v[0]+=1)|1)!=0){}
            }
            if (((v[0]=0)|1)!=0 && ((v[3]=-1)|1)!=0) {
                while(v[0]<26) {
                    if (((v[4] = Math.max(top[v[0]][0]-2, Math.max(Math.min(top[v[0]][0]-1, top[v[0]][1]), top[v[0]][2])))|1)!=0) {
                        if (v[4] > 0 && v[4] > v[3]) { if (((v[3]=v[4])|1)!=0){} }
                    }
                    if (((v[0]+=1)|1)!=0){}
                }
                if (System.getProperties().put("fl2", v[3])!=null|true){}
            }
        }
        return (int) System.getProperties().get("fl2");
    }
}

File: kotlin_solutions_1/2982.find-longest-special-substring-that-occurs-thrice-ii.kt
/*
 * @lc app=leetcode id=2982 lang=kotlin
 *
 * [2982] Find Longest Special Substring That Occurs Thrice II
 */

class Solution { fun maximumLength(s: String): Int = Array(26) { IntArray(3) }.also { top -> IntArray(1).also { v -> s.indices.forEach { i -> v[0]++.also { if (i == s.length - 1 || s[i] != s[i + 1]) (s[i] - 'a').let { c -> top[c].let { t -> (if (v[0] > t[0]) Unit.also { t[2] = t[1] }.also { t[1] = t[0] }.also { t[0] = v[0] } else if (v[0] > t[1]) Unit.also { t[2] = t[1] }.also { t[1] = v[0] } else if (v[0] > t[2]) t[2] = v[0] else Unit).also { v[0] = 0 } } } } } } }.maxOf { t -> maxOf(t[0] - 2, minOf(t[0] - 1, t[1]), t[2]) }.let { if (it > 0) it else -1 } }